benches/fibonacci.rs

```diff
@@ -6,135 +6,236 @@ use criterion::{
     BenchmarkId, Criterion, SamplingMode,
 };
 
-use pasta_curves::pallas;
-
 use lurk::{
-    circuit::circuit_frame::MultiFrame,
-    eval::{
-        empty_sym_env,
-        lang::{Coproc, Lang},
-    },
+    eval::lang::{Coproc, Lang},
     field::LurkField,
-    proof::nova::NovaProver,
-    proof::Prover,
-    ptr::Ptr,
+    proof::{nova::NovaProver, Prover},
     public_parameters::{
         instance::{Instance, Kind},
         public_params,
     },
     state::State,
-    store::Store,
 };
+use pasta_curves::pallas;
 
 const PUBLIC_PARAMS_PATH: &str = "/var/tmp/lurk_benches/public_params";
 
-fn fib<F: LurkField>(store: &Store<F>, state: Rc<RefCell<State>>, _a: u64) -> Ptr<F> {
-    let program = r#"
-(letrec ((next (lambda (a b) (next b (+ a b))))
-           (fib (next 0 1)))
-  (fib))
-"#;
-
-    store.read_with_state(state, program).unwrap()
-}
-
-// The env output in the `fib_frame`th frame of the above, infinite Fibonacci computation will contain a binding of the
-// nth Fibonacci number to `a`.
-// means of computing it.]
-fn fib_frame(n: usize) -> usize {
-    11 + 16 * n
+#[allow(clippy::upper_case_acronyms)]
+#[derive(Copy, Debug, Clone, PartialEq, Eq)]
+enum Version {
+    ALPHA,
+    LEM,
 }
 
-// Set the limit so the last step will be filled exactly, since Lurk currently only pads terminal/error continuations.
-fn fib_limit(n: usize, rc: usize) -> usize {
-    let frame = fib_frame(n);
-    rc * (frame / rc + usize::from(frame % rc != 0))
-}
-
-struct ProveParams {
-    fib_n: usize,
+pub struct ProveParams {
+    folding_steps: usize,
     reduction_count: usize,
+    version: Version,
 }
 
 impl ProveParams {
     fn name(&self) -> String {
+        format!("{:?},rc={}", self.version, self.reduction_count)
+    }
+}
+
+mod alpha {
+    use lurk::{circuit::circuit_frame::MultiFrame, eval::empty_sym_env, ptr::Ptr, store::Store};
+
+    use super::*;
+
+    fn fib<F: LurkField>(store: &Store<F>, state: Rc<RefCell<State>>) -> Ptr<F> {
+        let program = r#"
+    (letrec ((next (lambda (a b) (next b (+ a b))))
+               (fib (next 0 1)))
+      (fib))
+    "#;
+
+        store.read_with_state(state, program).unwrap()
+    }
+
+    pub fn prove<M: measurement::Measurement>(
+        prove_params: ProveParams,
+        c: &mut BenchmarkGroup<'_, M>,
+        state: Rc<RefCell<State>>,
+    ) {
+        let ProveParams {
+            folding_steps,
+            reduction_count,
+            version,
+        } = prove_params;
+
+        assert_eq!(version, Version::ALPHA);
+        let limit = reduction_count * folding_steps;
```

Comment:
Based on discussion elsewhere, I think this should be `reduction_count * (folding_steps + 1)`, right?

Reply:
Good catch!